cmd/auth-provider-gcp/provider/provider.go

```diff
@@ -0,0 +1,105 @@
+/*
+Copyright 2020 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+	http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package provider
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/cloud-provider-gcp/pkg/credentialconfig"
+	"k8s.io/cloud-provider-gcp/pkg/gcpcredential"
+	credentialproviderapi "k8s.io/kubelet/pkg/apis/credentialprovider"
+	"net/http"
+	"os"
+	"time"
+)
+
+const (
+	cacheDurationKey          = "KUBE_SIDECAR_CACHE_DURATION"
+	metadataHTTPClientTimeout = time.Second * 10
+	apiKind                   = "CredentialProviderResponse"
+	apiVersion                = "credentialprovider.kubelet.k8s.io/v1alpha1"
+)
+
+var (
+	noDurationGiven time.Duration = 0
+)
```

Comment:
Typically we don't define constants in go for zero values, we just inline it.

Reply:
Fixed